ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlTarget.java

```diff
@@ -66,6 +68,7 @@
   private static final String STATUS_PATH = "/status";
   private static final String KSQL_PATH = "/ksql";
   private static final String QUERY_PATH = "/query";
+  private static final String QUERY_STREAM_PATH = "/query-stream";
```

Comment:
What's up with this? We have had to handle transient queries for a long time, so I'm not seeing how this could have been missing.

Reply:
`/query` is the http 1.1 path, handled by `StreamedQueryResource`.  This is the endpoint we want to eventually deprecate.  It's what's currently handling queries for the other methods of `KsqlTarget`.  I've added a new method which uses the new endpoint `/query-stream` which has the benefit of not using the threadpool and being fully async.

Reply:
Thanks, yeah, I was just surprised that we're not already using the async endpoint.